File: frontend/src/app/models/task/dto/read-task-dto.ts
import {ReadUserDTO} from '../../user/dto/read-user-dto';

export interface ReadTaskDTO {
  id: string;
  title: string;
  description: string;
  status: string;
  createdBy: ReadUserDTO;
  createdAt: Date;
  updatedAt: Date;
  dueDate: Date;
}

File: backend/src/main/java/com/example/task_manager/task/dto/ReadTaskDTO.java
package com.example.task_manager.task.dto;

import lombok.Value;

import java.time.Instant;
import java.util.List;

@Value
public class ReadTaskDTO {
    String title;
    String description;
    String status;
    String createdBy;
    Instant createdAt;
    Instant updatedAt;
    Instant dueDate;
    List<String> assignedUsers;
}